src/node/desktop/src/main/gwt-callback.ts

```diff
@@ -902,10 +903,31 @@ export class GwtCallback extends EventEmitter {
     });
 
     ipcMain.handle('desktop_startup_error_info', async (event, varName: string) => {
+      if (varName == 'launch_failed') {
+        this.addMacOSVersionError();
+      }
       return resolveTemplateVar(varName, this.errorPageData);
     });
   }
 
+  addMacOSVersionError(): void {
+    if (platform() === 'darwin') {
+      const release_major = parseInt(release().substring(0, release().indexOf('.')));
+      // macOS 11.0 uses darwin 20.0.0
+      if (release_major < 20) {
+        let versionError = 'You are using an unsupported operating system.' +
```

Comment:
Ideally we'd make these strings localizable. We use `i18next` package for localizing Electron-specific strings. I don't know, offhand, if these status pages are localized at all, so unless they are I wouldn't worry about it now (but maybe open a tracking issue for future consideration).

Reply:
@gtritchie looking at https://github.com/rstudio/rstudio/blob/main/src/cpp/desktop/html/error.html, we're unfortunately not doing any localization for this page. 

@kevinushey I updated this to include the version (I manipulated my local code to make the error appear for a supported version for this screenshot):
![image](https://user-images.githubusercontent.com/5323711/217080684-b5644f54-95f2-40dc-89b1-03607f8bd948.png)

Reply:
Issue for localization: https://github.com/rstudio/rstudio/issues/12672